hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java

```diff
@@ -39,7 +39,12 @@
   private double threshold;
 
   @Option(names = {"-i", "--idleiterations"},
-      description = "Maximum consecutive idle iterations")
+      description = "Maximum consecutive idle iterations",
+      //idleiteration could be as follows:
+      // 0 ： indicate running infinitely
+      // positive number ： indicate the specific iterations
+      // -1 : user does not set this value
+      defaultValue = "-1")
```

Comment:
How about we use -1 as the value for running balancer infinitely? The possible value are >= -1 then. We don't need a default value.
Maybe internally in start balancer command it would be better to use Optional. So that you would know if user has specified an option like idleIteration 0. For input 0 I think the command should throw an exception because that is like not starting the balancer.

Reply:
yes , this seems better , will change this

Reply:
i think it would be a little complex to use optional here , since we just need to avoid an input idleiteration with 0 and this can be implemented by a simple judgement. what is more, optional is usually used to handle null(maybe not 0 or some other special values), so i think it is better not to use optional here, what do you think?

Reply:
How do you know user has entered 0? The variable will be initialised as 0 always.

Reply:
yes, this is indeed a problem, which i tried to assign a default value to solve it, but missed in the lastest commit. i see the doc [here](https://picocli.info/#_optionalt), i think here we should use an optional as you said. thanks, i will update

Reply:
i have update this PR , in which i replaced all the parameters with optional , PATL!